src/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -389,6 +391,14 @@ public function isTransient(string $className)
             $this->initialize();
         }
 
+        if (class_exists($className, false) && (new ReflectionClass($className))->isAnonymous()) {
```

Comment:
could this be on the hot path? strpos for `@` should be enough to check for anonymous classes if yes

Reply:
That's the same check than done in `getMetadataFor` already.